src/core/pos.ml

```diff
@@ -70,10 +70,14 @@ let in_pos : pos -> 'a -> 'a loc =
 let none : 'a -> 'a loc =
   fun elt -> { elt ; pos = None }
 
-(** [to_string pos] transforms [pos] into a readable string. *)
-let to_string : pos -> string = fun p ->
+(** [to_string ?print_fname pos] transforms [pos] into a readable string. If
+    [print_fname] is [true] (the default), the filename contained in [pos] is
+    printed. *)
+let to_string : ?print_fname:bool -> pos -> string =
+  fun ?(print_fname=true) p ->
   let {fname; start_line; start_col; end_line; end_col} = Lazy.force p in
   let fname =
+    if not print_fname then "" else
```

Comment:
You can avoid `not` by inverting branches.

Reply:
Yes, but this form allows the big part of the code to be in the `else` branch, and aligned with the `if`. I agree that introducing a negation can be a bit disturbing.